File: test/fixtures/ruby/get_charles_syntax.rb
require 'net/http'

uri = URI('http://localhost:28139/?format=json&')
req = Net::HTTP::Get.new(uri)
req['Host'] = 'api.ipify.org'
req['Accept'] = '*/*'
req['User-Agent'] = 'GiftTalk/2.7.2 (iPhone; iOS 9.0.2; Scale/3.00)'
req['Accept-Language'] = 'en-CN;q=1, zh-Hans-CN;q=0.9'

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/go/get_charles_syntax.go
package main

import (
	"fmt"
	"io"
	"log"
	"net/http"
)

func main() {
	client := &http.Client{}
	req, err := http.NewRequest("GET", "http://localhost:28139/?format=json&", nil)
	if err != nil {
		log.Fatal(err)
	}
	req.Header.Set("Host", "api.ipify.org")
	req.Header.Set("Accept", "*/*")
	req.Header.Set("User-Agent", "GiftTalk/2.7.2 (iPhone; iOS 9.0.2; Scale/3.00)")
	req.Header.Set("Accept-Language", "en-CN;q=1, zh-Hans-CN;q=0.9")
	resp, err := client.Do(req)
	if err != nil {
		log.Fatal(err)
	}
	defer resp.Body.Close()
	bodyText, err := io.ReadAll(resp.Body)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("%s\n", bodyText)
}
